extensions/amp-subscriptions/0.1/entitlement.js

```diff
@@ -101,6 +102,9 @@ export class Entitlement {
    * @return {boolean}
    */
   enablesThis() {
+    if (!this.product_) {
+      dev().warn('amp-subscriptions', 'Current Product is not set');
```

Comment:
SMH, this feels wrong. This should be a dev assert - i.e. if this happens, this is just our bug.